core/repos/repomanager.go

```diff
@@ -134,21 +145,81 @@ func (manager *RepoManager)  discoverRepos() (repos []string, err error){
 	return
 }
 
-func (manager *RepoManager) AddRepo(repoPath string) error {
+func (manager *RepoManager) checkDefaultRevision(repo *Repo) error {
+	_, err := git.PlainClone(repo.getCloneDir(), false, &git.CloneOptions{
+		URL:    repo.getUrl(),
+		ReferenceName: plumbing.NewBranchReferenceName("master"), //Check out master so we have a working copy
+	})
+
+	if err != nil && err != git.ErrRepositoryAlreadyExists {
+			return err
+	}
+
+	// Decide if revision = defaultRevision -> if yes lock them together
+	// We do this because, in the case where the revision was not specified, it would fall back to the default revision
+	// As a result, subsequent changes to the default revision should be followed by revision
+	revIsDefault := false
+	if repo.DefaultRevision == repo.Revision {
+		revIsDefault = true
+	}
+
+	// Check that the defaultRevision is valid
+	var prefixes []string
+	prefixes = append(prefixes, refRemotePrefix, refTagPrefix)
+	var matchedRevs []string
+	matchedRevs, err = repo.getRevisions(repo.DefaultRevision, prefixes)
+	if err != nil {
+		return err
+	} else if len(matchedRevs) == 0 {
+		log.Warning("Default revision " + repo.DefaultRevision + " invalid for " + repo.GetIdentifier())
+		if repo.DefaultRevision != manager.defaultRevision { //TODO: This check implies that repo.DefaultRevision might be different to the global on. This is not the case yet but it stays here for reference.
```

Comment:
Not sure I understand the TODO. You mean that it will be possibly different from the global default once we load persistent per-repo defaults?